### DISCUSSION TITLE
Cobbler3.3.4 Compiling RPM Package

### DISCUSSION THREAD

**Discussion by 1392273211** [cobbler/cobbler#3796] [Q&A]
When can the official website support the rpm package for cobbler 3.3.0 to 3.3.6, or write a documentation for compiling the source code into an rpm package? I couldn't find detailed compilation documentation online, which prevented me from using the latest cobbler

**Comment by SchoolGuy** [comment]
The documentation is up and running:

- https://cobbler.readthedocs.io/en/v3.3.6/
- https://cobbler.readthedocs.io/en/v3.3.5/
- ...

That also contains the installation guide.

**Reply by 1392273211** [reply]
I compiled it based on this document, but the compilation failed and I don't know how to solve it

**Reply by SchoolGuy** [reply]
Any error messages?

**Reply by 1392273211** [reply]
I tried to compile 3.2.2, the operation flow
wget https://github.com/cobbler/cobbler/archive/refs/tags/v3.2.2.tar.gz
pip3 install -i https://mirrors.aliyun.com/pypi/simple/ sphinx
pip3 install -i https://mirrors.aliyun.com/pypi/simple/ coverage
yum -y install git make gcc

[root@localhost cobbler-3.2.2]# cat /etc/os-release
NAME="Rocky Linux"
VERSION="8.6 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.6 (Green Obsidian)"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky Linux"
ROCKY_SUPPORT_PRODUCT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8"
[root@localhost cobbler-3.2.2]#


Report an error

[root@localhost cobbler-3.2.2]# make rpms
cleaning: python bytecode
cleaning: build artifacts
cleaning: temp files
No pyflakes found
No pycodestyle found
creating: AUTHORS
fatal: not a git repository (or any of the parent directories): .git
creating: sdist
ERROR, unknown distro
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_wheel'
make: *** [Makefile:88: sdist] Error 1
[root@localhost cobbler-3.2.2]#

**Reply by SchoolGuy** [reply]
@1392273211 This is an issue by the `distro_build_configs.sh` script that fails if you are running on a distro that Cobbler isn't supported to run on. For 3.2.2 the support for building on Rocky Linux wasn't present, as such this will not work. A new version of Cobbler - 3.2.3 - will be released soon, I will ensure that a fix for this will be present.

**Reply by beep-psh** [reply]
My way to get around that specific error was solved by running <code>pip3 install wheel</code> and my source was: [this](https://stackoverflow.com/questions/37029385/cant-build-wheel-error-invalid-command-bdist-wheel)
